src/atoms/referencesState.ts

```diff
@@ -0,0 +1,31 @@
+import { atom } from 'jotai';
+
+import { ReferenceItem } from '../types/ReferenceItem';
+
+const pdfIngestionProcessAtom = atom<string[]>([]); // ids of pending processes for now, this could be a record with progress percentage in the future
+
+export const isPdfIngestionRunningAtom = atom((get) => get(pdfIngestionProcessAtom).length > 0);
+
+export const addPdfIngestionProcessAtom = atom(null, (_get, set, value: string) => {
+  set(pdfIngestionProcessAtom, (pendingProcesses) => [...pendingProcesses, value]);
+});
+
+export const removePdfIngestionProcessAtom = atom(null, (_get, set, value: string) => {
+  set(pdfIngestionProcessAtom, (pendingProcesses) => pendingProcesses.filter((processId) => processId !== value));
+});
+
+type ReferencesState = Record<string, ReferenceItem>;
+
+const referencesAtom = atom<ReferencesState>({});
+
+export const getReferencesAtom = atom((get) => Object.values(get(referencesAtom)));
+
+export const getDerivedReferenceAtom = (referenceId: string) => atom((get) => get(referencesAtom)[referenceId]);
+
+export const setReferencesAtom = atom(null, (_get, set, references: ReferenceItem[]) => {
+  const updatedReferences = {} as ReferencesState;
+  references.forEach((reference) => {
+    updatedReferences[reference.id] = reference;
+  });
+  set(referencesAtom, updatedReferences);
+});
```

Comment:
I'm a bit confused by this write-only atom.
If the idea is to set the `referencesAtom` why can't one use the atom directly instead of via a write-only atom.

Maybe we want to read the existing `referencesAtom` and use that as initial value (line 26). Then this would be a _patial update_ for the references atoms.

Reply:
I agree this could be `addReferencesAtom`, so we keep the current state as default value and add the new values. I did it this way because the python process returns everything anyway, but that is a bit weird, I will update it.

About the write-only atom is because the internal way of storing references is different from the way components use them: it is convenient to have an object to store them, so that we can get any reference by id in O(1) (or maybe this should be a `Map`, I'm happy to change it), but then the components handle arrays of references, not objects. When the ingestion process is done, it returns an array, then it shouldn't be the component's job to transform this array into the structure we use in the atom, hence this write-only atom. Here again, decoupling internal storage from getter/setter allows us to change the internal structure any time without having to update the code everywhere.